db/db_with_timestamp_basic_test.cc

```diff
@@ -3918,6 +3918,91 @@ TEST_F(DBBasicTestWithTimestamp, IterSeekToLastWithIterateUpperbound) {
   ASSERT_FALSE(iter->Valid());
   ASSERT_OK(iter->status());
 }
+
+TEST_F(DBBasicTestWithTimestamp, TimestampFilterTableReadOnGet) {
+  Options options = CurrentOptions();
+  options.env = env_;
+  options.create_if_missing = true;
+  options.statistics = ROCKSDB_NAMESPACE::CreateDBStatistics();
+  const size_t kTimestampSize = Timestamp(0, 0).size();
+  TestComparator test_cmp(kTimestampSize);
+  options.comparator = &test_cmp;
+  BlockBasedTableOptions bbto;
+  bbto.block_size = 100;
+  options.table_factory.reset(NewBlockBasedTableFactory(bbto));
+  DestroyAndReopen(options);
+
+  // Put
+  // Create two SST files
+  // file1: key => [1, 2], timestamp => [10, 20]
```

Comment:
Hmm. This is odd. I would expect your table filtering code never to be reached on file2 because we only query a file if the target key is within the range of its smallest to largest keys (which is recorded outside the file itself).

It looks like you need at least 3 files for that to work:
https://github.com/facebook/rocksdb/blob/8.1.fb/db/version_set.cc#L187

So (not your fault) I don't like that this is relying on an obscure heuristic to actually test what it is intended to test. It would be better to use file1 keys {1, 3} and file2 keys {2, 4}, and have the Gets use key 3.